### PR TITLE
Combat Stuff: Monk & Rogue's Delight

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About The Pull Request

adds headbutting and hostage taking as a mechanic

to hostage take:
1. make sure target is not in cmode
2. grip enemy throat
3. upgrade grab
4. use threat intent on enemy
5. now if they try to resist or move away they get cut in the throat (hitting throat not guaranteed)

to head butt:
1. grab enemy head
2. click yourself with grab item
3. they are now headbutted
4. good job

kicking also has a smaller cooldown and you can stomp someones throat

## Why It's Good For The Game

COMBAT MECHANICS!!! AGRH!!!!

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/modules/mob/living/grabbing.dm [hunk 1]
```diff
@@ -149,6 +168,22 @@
...
+		if(/datum/intent/grab/hostage)
+			if(limb_grabbed && grab_state > 0) //this implies a carbon victim
+				if(ishuman(M) && M != user)
+					var/mob/living/carbon/human/H = M
+					var/mob/living/carbon/human/U = user
+					if(U.cmode)
+						if(H.cmode)
+							to_chat(U, "<span class='warning'>[H] is too prepared for combat to be taken hostage.</span>")
+							return
+						to_chat(U, "<span class='warning'>We take [H] hostage.</span>")
```
Comment (ThePainkiller) [inline code/modules/mob/living/grabbing.dm:180]:
```suggestion
						to_chat(U, "<span class='warning'>I take [H] hostage.</span>")
```
_"We"_ are not changelings, first person would be best

**Review by ThePainkiller — DISMISSED** [review]

File: code/modules/mob/living/grabbing.dm (hunk 1 above)
Comment (casualspacestation14enjoyer) [reply code/modules/mob/living/grabbing.dm:180]: This has been an issue for me for a long... LONG time. I cannot stress how long it took me to finally decide on 'we'. Thanks.

**Comment by ThePainkiller** [comment]
Gonna give this a spin on testmerging to see how well it goes, and if it needs adjustments or is unfair (fearing that people with nosegrab will abuse this)

**Comment by NPC1314** [comment]
No reports of issues, time to fullmerge methinks?

**Review by ThePainkiller — APPROVED** [review]
At this point no issues were reported and this would add good flavor for CQC combat.

**Comment by Ickysoup** [comment]
If combat mode is enabled after getting taken hostage, will it matter?

**Pull request merged**